src/executables/util_cs.c

```diff
@@ -3993,3 +3993,255 @@ tde (UTIL_FUNCTION_ARG * arg)
 error_exit:
   return EXIT_FAILURE;
 }
+
+static time_t
+parse_date_string_to_time (char *date_string)
+{
+  assert (date_string != NULL);
+
+  time_t result = 0;
+  struct tm time_data = { };
+
+  if (sscanf
+      (date_string, "%d-%d-%d:%d:%d:%d", &time_data.tm_mday, &time_data.tm_mon, &time_data.tm_year, &time_data.tm_hour,
+       &time_data.tm_min, &time_data.tm_sec) != 6)
+    {
+      return 0;
+    }
+
+  if (time_data.tm_mday < 1 || time_data.tm_mday > 31
+      || time_data.tm_mon < 1 || time_data.tm_mon > 12
+      || time_data.tm_year < 1900
+      || time_data.tm_hour < 0 || time_data.tm_hour > 23
+      || time_data.tm_min < 0 || time_data.tm_min > 59 || time_data.tm_sec < 0 || time_data.tm_sec > 59)
+    {
+      return 0;
+    }
```

Comment:
Yes, you are right, it can not filter like 2/30, 4/31, ..etc. 
we don't have to validate the time string, but I just follow the spec of the members in `struct tm` which is used to get `time_t` value.